src/ServiceControl.Audit/Infrastructure/RavenDB/EmbeddedRavenDbHostedService.cs

```diff
@@ -16,9 +16,14 @@ class EmbeddedRavenDbHostedService : IHostedService
         readonly IEnumerable<IDataMigration> dataMigrations;
 
         public EmbeddedRavenDbHostedService(IDocumentStore documentStore, IOptions<RavenStartup> ravenStartup, IEnumerable<IDataMigration> dataMigrations)
+        : this(documentStore, ravenStartup.Value, dataMigrations)
+        {
+        }
+
+        public EmbeddedRavenDbHostedService(IDocumentStore documentStore, RavenStartup ravenStartup, IEnumerable<IDataMigration> dataMigrations)
         {
             this.documentStore = documentStore;
-            this.ravenStartup = ravenStartup.Value;
+            this.ravenStartup = ravenStartup;
```

Comment:
I noticed below that the IDocumentStore instance is disposed of when this service stops, but this service didn't initialize the store.  Should the disposal be handled by the DI container or the calling code instead?

Reply:
It kind of does. An `IDcoumentStore` instance is unusable before you call `.Initialize()` on it, which we do in our startup. Everything before that is effectively configuration.

We struggled with the right way to do this when we introduced the hosted service and this was the only way we found that works.